### ISSUE TITLE
iOS and Android App

### ISSUE THREAD

**Issue by saivan** [typora/typora-issues#525]
It seems that typora is built on something like electron if I'm not mistaken. Why not make it into an ios app so that it can be used in a mobile setting. I don't know how hard this would be, but I'd happily pay for a version of typora that works with iOS since it would be easy to get handwritten notes into it. Perhaps use something like:

https://www.nativescript.org/

**Comment by jdseyres** [comment]
I'd really love to have an iOS version too, both on iPhone and iPad.
And I'd be ready to pay for it too.

**Comment by allanfish** [comment]
typora is so good,  I am waiting for android version!

**Comment by blankmann** [comment]
iOS Typora!  Please.

**Comment by x7x7** [comment]
I really expect an iOS version .

At present, no any markdown editor in iOS gives a satisfying latex function. If Typora-ios comes up, it will be the only one.

**Comment by davemacaulay** [comment]
I'd really love to see this, I thought Bear might work as well as Typora but it doesn't. 

There's some great solutions for building mobile apps using similar things to Electron:
https://ionicframework.com
https://facebook.github.io/react-native/

**Comment by vassudanagunta** [comment]
It would be great if the iOS version implemented iOS features that enable interoperability with [Working Copy](https://workingcopyapp.com) for Git and GitHub access, [Editorial](http://omz-software.com/editorial/) for publishing workflows, etc.

**Comment by HADB** [comment]
+1

**Comment by devcline** [comment]
+1 for ios version. I will gaveup a dropbox paper. Promise.

**Comment by somaGFX** [comment]
I really would love to see a Android and iOS version because I am heavily using both systems.

Typora is so good. Don't know why it's not open source though. It's made of a lot of open source stuff.

How ever I am a huge fan of the tool and use it since the first beta version came out.

**Comment by samirelanduk** [comment]
Would like to add my voice to the calls for an iOS version. No other app on iOS comes close to the functionality that the Desktop Typora app offers. Would gladly pay outright or via subscription for an iOS app.

**Comment by vassudanagunta** [comment]
In the meantime, checkout http://1writerapp.com. It syncs via both iCloud and Dropbox, has **full text search**, **indexes #hashtags**. automatically imports and applies a customizable template to selected text from other apps (e.g. Safari) via share sheet, and more...

**Comment by boettges** [comment]
@vassudanagunta Thanks for the tip! How do you work with hashtags in Typora? For me they always get converted to headlines

**Comment by vassudanagunta** [comment]
Yeah that would be a big change for Typora, because right now it really is an editor for _individual_ files, not _assemblies_ of pages. For starters, as you said, there would have to be a root folder.

I think it is important for Typora to focus on getting rock solid about doing individual pages. Once it gets there, I hope it supports what 1Writer does, what you suggest. 

There is no standard for **assemblies of markdown content**. If you have a folder of Markdown content that works in 1Writer, it doesn't work as-is in Jekyll or Hugo. I'm actually trying to address that with [Markaround](https://github.com/Markaround/markaround). With a standard, you could have your folder hierarchy of notes and it would be portable across editors, static-site generators, and support a lot of cool features like automatic broken link detection or automatic link updating when you move or rename a page. If you want to be updated, star/watch the repo!  :)

**Comment by boettges** [comment]
Actually my most used feature of Typora is to drag the "root"-folder of my notes into the window and have Typora list all .md-files in it. I navigate between them in a similar fashion as I would with notes in other note-taking applications, but just on file-system basis.

Markaround looks like a solid idea as well 👌

**Comment by vassudanagunta** [comment]
I love that Typora does that. But to support an index of hashtags for all content within a given folder, it would have to maintain an index specific to that folder:

- the index would include only documents below that folder
- the paths to all files in the index would have to be relative to that folder

The only way Typora could do this far any arbitrary folder that you drag into it:

- create these indexes *on-the-fly in memory* every time you open/drag a folder into it. I'm not sure that will scale to a folder containing hundreds of files. 
- create these indexes and cache them in the folder in a hidden folder (e.g. `.Typora`). It would do this for every folder you opened/dragged as a root for you current work. This would lead to such indexes littering your file system, including within the same document tree if you happen to open subfolders as the root. There would be a significant performance impact every time Typora indexes a new root.
- support explicit definition of fixed "project" or "document collection" folders. This is what I'm sure most note taking apps with indexes do. 

Regardless, it is a lot more involved than what Typora does now, which is simply to show the folder tree in the sidebar, without giving any special treatment to that tree or its root folder. It a big change!

But *IF* Typora one day did this, it could have so many benefits:

- You could use absolute paths for internal links within that document tree, e.g.:
    ```
    [link](/section/subsection/topic.md)
    ```
- You could define themes specific to a document tree.
- Typora could automatically update links if you moved or renamed files (This would be amazing).
- And of course the ability to index all the content, for either full text search (#651) or #hashtags or both. 

@abnerlee seemed to close the door to this idea a couple of years ago (#57), or maybe he was just deferring it?

My goal for [Markaround](https://github.com/Markaround/markaround) is to make document trees or *assemblies* standardized and portable. It would allow doing all of the above in a portable way.

**Comment by LaputaGit** [comment]
1+ for ios

**Comment by Philiphlop** [comment]
@vassudanagunta 

> create these indexes on-the-fly in memory every time you open/drag a folder into it. I'm not sure that will scale to a folder containing hundreds of files.

This is how VSNotes does it. An extension for Visual Studio Code, which reads the front matter of all notes in a loaded directory, then builds a tag list. It worked alright, but there is a lag on first load as you mention.

**Comment by charleshan** [comment]
Any recommendations for using Typora + Google Drive on Android?

**Comment by rdewolff** [comment]
Any news ?

**Comment by charleshan** [comment]
I started using Dropbox Paper because it has a pretty good markdown export.

**Comment by zkelo** [comment]
+1 for android

**Comment by nobility1999** [comment]
+1 for ios, please!!!!

**Comment by dtvrs** [comment]
+1 for Android, typora easily beats Textie, which is great, but typora please 😁

**Comment by vpontin** [comment]
+1 for Android, the only alternative i found for now is [StackEdit](https://stackedit.io), but it lacks WYSIWYG-ness Typora has (and isnt a native app, so try to went offline).

**Comment by JingwenLiang** [comment]
+1 for ios app！！！！！

**Comment by Philiphlop** [comment]
Please don't comment with +1, it only spams up the issue. Reserve comments for feedback and input into the feature. Instead please thumbs up the issue, so that the dev can see the interest from an analytical point of interest. 

![screenshot from 2018-10-03 07-17-51](https://user-images.githubusercontent.com/22427598/46393319-8e75f500-c6dc-11e8-9798-c820079ab410.png)

**Comment by vassudanagunta** [comment]
@abnerlee maybe it makes sense to `Lock conversation` since no one can really supply more information and we're all getting spam in our inbox. You can add a final message before locking it, such as:

> Comments are locked. If you want to express your approval or disapproval, please add your reaction at the top.

**Comment by vpontin** [comment]
I think the folks think without constant "updating" the issue the devs will just forget about it.....

**Comment by abnerlee** [comment]
I will lock this since this gets too long. 

We fully understand your requirements, for now, we can only focus on the desktop version.

Note: Typora is a Markdown editor, so you could still choose any Markdown/Text editor to work with Typora on mobile devices.
